release-notes: added release notes for v4.6.0 (#49)
bisdn/bisdn-linux: release-notes.txt

BISDN Linux
v4.6.0 Release Notes
Date: 2022-05-03

> ! New in this release

Reviewer: We have a new OF-DPA option for controlling memory allocation for Layer 2 and Layer 3 tables which should be mentioned here.

Also it would be good to thematically group things, so have OF-DPA things together.

> - New platform Accton-AS5835-54x is now supported

Reviewer 2: ```suggestion
- New platform Accton AS5835-54x is now supported
```

- baseboxd: bump version to 1.11.2

> - Licenses in build process were updated to use the correct SPDX identifier

Author: Removed from notes

- Linux kernel updated to version 5.15.35

> - OF-DPA updated to support configuraton of MPLS pseudowires using RYU

Reviewer: is this the correct way? AFAIU Ryu now supports configuring MPLS pseudowires in OF-DPA, not the other way around

- FRR updated to version 8.2

> - baseboxd/OF-DPA: support added for creating and enabling Broadcom KNET interfaces

Reviewer: This was already part of v4.5.0.

Author: This was 'experimental' in v4.5.0, should I reword it somehow?

Author: Updated the text to say '.. now fully supported'

> ! Bug fixes

Author: added lines to both of these in bug fixes

> - sysctl max read buffer size was set lower than the defaul read buffer size, causing baseboxd to intitialize with a buffer size that is too small. This could lead to loss of netlink messages, causing an out of sync switch state

Reviewer 2: Seems a bit long. How about:
```suggestion
- increase max read buffer to avoid loss of netlink messages (and out of sync switch state) under heavy load
```

- Broadcom-KNET interfaces: now properly generates random MAC addresses
- Broadcom-KNET interfaces: fixed issue related to race condition upon interface creation
- baseboxd: broadcom-KNET interfaces are now properly removed when stopping baseboxd

> - client_drivshell: Fixed an issue when running that caused client_drivshell to segfault when running 'client_drivshell log'

Reviewer 2: ```suggestion
- client_drivshell: Fixed segfault when running 'client_drivshell log'
```

Changes between v4.5.0 and v4.6.0:

bisdn-onie-additions:
mk_bisdn_yocto_image.sh: add note on legacy status
mk_bisdn_yocto_image: document usage
mk_bisdn_yocto_image: document source
mk_onie2: add rudimentary code documentation
accton-as4610/platform.conf: add coments
accton-as4610/platform.conf: document source
u-boot-arch/install.sh: add comments
u-boot-arch/install.sh: document source
x86_64/install.sh: document source
installer/install.sh: document functions
installer/install.sh: add comments
installer: accton-as4610/platform.conf: fix default configuration
installer: fix detect_bisdn_linux_msdos_partition
installer: fix create_bisdn_linux_msdos_partition args
accton-as5835-54x: add platform.conf

build-bisdn-linux:
accton-as5835-54x: add platform to local.conf.sample

> release-notes: added release notes for v4.5.0 (#44)

Reviewer 2: While technically true, we may want to drop this line.

Author: I removed it

BISDN Linux
v4.6.0 Release Notes
Date: 2022-05-04

! New in this release

> - New platform Accton AS5835-54x is now supported

Author: Added the following line:
`- New platform Edgecore DCS201 (AS5835-54X) is now supported. The DCS201 is a
7     Trident 3 with 8x 10GbE SFP+ ports and 6x 100GbE QSFP28 ports`

- OF-DPA extensions for ryu and sample code added for configuring MPLS
  pseudowires
- OF-DPA: new option for controlling memory allocation for Layer 2 and Layer 3
  tables
- baseboxd/OF-DPA: Broadcom KNET interfaces are now fully supported
- baseboxd: bump version to 1.11.2
- Linux kernel updated to version 5.15.35
- FRR updated to version 8.2

! Bug fixes

> - increase max read buffer to avoid loss of netlink messages (and out of sync

Reviewer: ```suggestion
- increased max read buffer to avoid loss of netlink messages (and out of sync
```

Author: Correct tense

  switch state) under heavy load

> - Added missing md5 authentication support on Accton AS4610

Reviewer: ```suggestion
- AS4610: added missing Kernel option to allow BGP MD5 authentication
```

BISDN Linux
v4.6.0 Release Notes
Date: 2022-05-04

! New in this release
- New platform Edgecore DCS201 (AS5835-54X) is now supported. The DCS201 is a

>   Trident 3 with 8x 10GbE SFP+ ports and 6x 100GbE QSFP28 ports

Reviewer: pretty sure this should be

```suggestion
  Trident 3 with 48x 10GbE SFP+ ports and 6x 100GbE QSFP28 ports
```

Author: Good catch, my bad

  pseudowires
- OF-DPA: new option for controlling memory allocation for Layer 2 and Layer 3
  tables
- baseboxd/OF-DPA: support added for creating and enabling Broadcom KNET
  interfaces

> - baseboxd: bump version to 1.11.2

Author: agreed

- Linux kernel updated to version 5.15.35

> - FRR updated to version 8.2

Author: I remember we had some talks about this last month regarding 8.2 or 8.2.2, Official page and github says 8.2.2 so this should be our announcement as well.

! Bug fixes
- increased max read buffer to avoid loss of netlink messages (and out of sync
  switch state) under heavy load
- AS4610: added missing Kernel option to allow BGP MD5 authentication
- EPS202 (AS4630-54PE): enabled the LM77 driver, fixing reading out temperatures from
  the fourth sensor in onlp
- Broadcom-KNET interfaces: now properly generates random MAC addresses

> - Broadcom-KNET interfaces: fixed issue related to race condition upon
>   interface creation

Reviewer: ```suggestion
- Broadcom-KNET interfaces: fixed a race condition on interface creation that could cause system hangs when configuring interfaces using systemd-networkd
```

(add line breaks as needed)

Author: Updated line with the nessesary line break
